app/main.py

```diff
@@ -1,6 +1,19 @@
 from typing import Callable
 
 
+cache_history = dict()
```

Comment:
This variable must be inside decorator, to create separate state for each function